src/reanimated2/screenTransition/commonTypes.ts

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+import type {
+  MeasuredDimensions,
+  ShadowNodeWrapper,
+  SharedValue,
+} from '../commonTypes';
+
+export type PanGestureHandlerEventPayload = {
+  x: number;
+  y: number;
+  absoluteX: number;
+  absoluteY: number;
+  translationX: number;
+  translationY: number;
+  velocityX: number;
+  velocityY: number;
+};
+
+export type AnimatedScreenTransition = {
+  topScreenFrame: (
+    event: PanGestureHandlerEventPayload,
+    screenSize: MeasuredDimensions
+  ) => Record<string, unknown>;
+  belowTopScreenFrame: (
+    event: PanGestureHandlerEventPayload,
+    screenSize: MeasuredDimensions
+  ) => Record<string, unknown>;
+};
+
+export type GoBackGesture =
+  | 'swipeRight'
+  | 'swipeLeft'
+  | 'swipeUp'
+  | 'swipeDown'
+  | 'verticalSwipe'
+  | 'horizontalSwipe'
+  | 'twoDimensionalSwipe';
+
+export type ScreenTransitionConfig = {
+  stackTag: number;
+  belowTopScreenId: number | ShadowNodeWrapper;
+  topScreenId: number | ShadowNodeWrapper;
+  screenTransition: AnimatedScreenTransition;
+  sharedEvent: SharedValue<PanGestureHandlerEventPayload>;
+  startingGesturePosition: SharedValue<PanGestureHandlerEventPayload>;
+  onFinishAnimation?: () => void;
+  isTransitionCanceled: boolean;
```

Comment:
Just wanted to confirm if we choose to use form "canceled" or "cancelled".
```suggestion
  isTransitionCancelled: boolean;
```